pytorch_lightning/trainer/trainer.py

```diff
@@ -358,7 +358,9 @@ def __init__(
         self.on_tpu = tpu_cores is not None
         self.tpu_cores = tpu_cores
         assert self.tpu_cores in (1, 8, None) or (
-            isinstance(self.tpu_cores, (list, tuple, set)) and len(self.tpu_cores) == 1
+            isinstance(self.tpu_cores, (list, tuple, set)) and
+            len(self.tpu_cores) == 1 and
+            self.tpu_cores[0] in range(1, 9)
         ), '`tpu_cores` can only be 1, 8 or [<1-8>]'
```

Comment:
this is confusing [<1-8>] is not clear.

What are the allowed values now?
how do you differentiate between 1 tpu core and tpu core index 1?

for gpus it's
```
--gpus 1
--gpus '1,'
```

Reply:
I think for 1 tpu core it is `tpu_cores=1` and for a specific tpu core it iss `tpu_cores=[tpu_core_idx]`.

Reply:
@williamFalcon we differentiate the following way:
```
Trainer(tpu_cores=1) # 1 core with MP
Trainer(tpu_cores=8) # 8 core with MP
Trainer(tpu_cores=[2]) # Single core (index 2) without MP
Trainer(tpu_cores=[6]) # Single core (index 6) without MP
```

Reply:
we can mock the argparse and test directly these arguments, see
https://github.com/PyTorchLightning/pytorch-lightning/blob/34237cfcaf5c53a15e62a3219d72bf33b667d214/tests/trainer/test_trainer_cli.py#L106

Reply:
@williamFalcon @Borda I have added the support for CLI arguments.
```
--tpu_cores 1
--tpu_cores '1,'
```
https://github.com/PyTorchLightning/pytorch-lightning/blob/62968c058dea1cfd20a4baf2e3f7cdd1754165b9/tests/trainer/test_trainer_cli.py#L104